src/main/java/org/qmstr/gradle/QmstrPackTask.java

```diff
@@ -36,13 +41,44 @@ void pack() {
                 .forEach(c -> c.getAllArtifacts().forEach(art -> arts.put(art, c.getResolvedConfiguration().getFiles())));
 
         bsc.SendBuildFileNodes(arts.entrySet().parallelStream()
-                .map(artEntry -> FilenodeUtils.processArtifact(artEntry.getKey(), artEntry.getValue()))
+                .map(artEntry -> processArtifact(artEntry.getKey(), artEntry.getValue()))
                 .filter(o -> o.isPresent())
                 .map(o -> o.get())
                 .collect(Collectors.toSet())); 
         
     
     }
+
+    public static Optional <Datamodel.FileNode> processArtifact(PublishArtifact artifact, Set<File> dependencySet) {
```

Comment:
most of this is library functionality
only PublishArtifact needs to be processed here.